### PR TITLE
Script to apply template to translations

### PR DESCRIPTION
**Pull request by adaext**
Replace the translations in dymmy template file with specified translation file

### CODE REVIEW

File: template.vim
Comment (flyingdragonfly) [inline template.vim]: As discussed offline, can we have this file deleted?

File: menu_trans.vim.template
Comment (flyingdragonfly) [inline menu_trans.vim.template]: Remove this line, as this template is supposed to be used in all languages.

File: menu_trans.vim.template
Comment (flyingdragonfly) [inline menu_trans.vim.template]: Move this into previous "maintainers", and add your name here.

File: menu_trans.vim.template
Comment (flyingdragonfly) [inline menu_trans.vim.template]: Update this date.

File: menu_trans.vim.template
Comment (flyingdragonfly) [inline menu_trans.vim.template:8]: Add a few lines of comments to describe that the generated translation files are creating using your tool. Provide link to your GitHub repository.

File: menu_trans.vim.template
Comment (flyingdragonfly) [inline menu_trans.vim.template:352]: Add empty line before this line.

File: menu_trans.vim.template
Comment (flyingdragonfly) [inline menu_trans.vim.template]: `menu.vim` is not a new file, but the main part of the whole translation. You should find proper place to insert these three missing translations, not a new section.

File: vim_menutrans_merger.py
Comment (flyingdragonfly) [inline vim_menutrans_merger.py]: Re-wrap your file to make sure no line is longer than 80 characters.

File: vim_menutrans_merger.py
Comment (flyingdragonfly) [inline vim_menutrans_merger.py]: Add a few spaces between ":" and "the" to align the descriptions vertically.

File: vim_menutrans_merger.py
Comment (flyingdragonfly) [inline vim_menutrans_merger.py]: Missing line break at the end of this file.

File: vim_menutrans_merger.py
Comment (flyingdragonfly) [inline vim_menutrans_merger.py]: Python uses `snake_case` to name the variables, not `camelCase`.

File: vim_menutrans_merger.py
Comment (flyingdragonfly) [inline vim_menutrans_merger.py]: Ditto. Should use `snake_case` naming rule.

File: vim_menutrans_merger.py
Comment (flyingdragonfly) [inline vim_menutrans_merger.py]: Always write to `test.vim`? We need to overwrite the original translation file with the merged result.

File: vim_menutrans_merger.py
Comment (flyingdragonfly) [inline vim_menutrans_merger.py]: Nit: this script is called `vim_menutrans_merger.py`. You probably need another pull request to fix the other script, too.

File: vim_menutrans_merger.py
Comment (flyingdragonfly) [inline vim_menutrans_merger.py]: `UTF-8`? You're going to fail a lot of non-UTF-8 files.

File: vim_menutrans_merger.py
Comment (flyingdragonfly) [inline vim_menutrans_merger.py]: Why print the line number out in your generated files? The result will not be a valid Vim script if you do that.

**Review by flyingdragonfly — CHANGES_REQUESTED** [review]

File: vim_menutrans_merger.py
Comment (adaext) [reply vim_menutrans_merger.py]: But we only care about UTF-8 files, because other different coded format files are generated from UTF-8 files.

File: vim_menutrans_merger.py
Comment (adaext) [reply vim_menutrans_merger.py:87]: ok, I will

Comment (adaext) [reply vim_menutrans_merger.py]: > 

ok, I will.

File: menu_trans.vim.template
Comment (flyingdragonfly) [inline menu_trans.vim.template]: Should keep "Yuheng Xie <elephant@linux.net.cn>" in "previous maintainers", too.

File: menu_trans.vim.template
Comment (flyingdragonfly) [inline menu_trans.vim.template:9]: Should have an empty line before this line.

File: menu_trans.vim.template
Comment (flyingdragonfly) [inline menu_trans.vim.template]: No, this comment is not supposed to be talking about the "template" file, but the generated final translations. Your template will be applied to all the existing translations, and this comment will be copied over verbatim. Therefore you should tell the future maintainer that the translation result was generated with your script.

File: vim_menutrans_merger.py
Comment (flyingdragonfly) [inline vim_menutrans_merger.py]: Why? You're still generating the line number in the final result, and then your final result will no longer be a valid Vim script.

File: vim_menutrans_merger.py
Comment (flyingdragonfly) [inline vim_menutrans_merger.py]: Why do you need `line_number`? I don't see it in use anywhere.

File: vim_menutrans_merger.py
Comment (flyingdragonfly) [inline vim_menutrans_merger.py]: Ditto. Is `line_number` actually in use?

File: vim_menutrans_merger.py
Comment (flyingdragonfly) [inline vim_menutrans_merger.py]: Are you sure Python allows you to wrap the line this way?

File: vim_menutrans_merger.py
Comment (flyingdragonfly) [inline vim_menutrans_merger.py]: Ditto. Line wrapping looks invalid.

File: vim_menutrans_merger.py
Comment (flyingdragonfly) [inline vim_menutrans_merger.py]: Ditto. Have you tested that it still works?

**Review by flyingdragonfly — CHANGES_REQUESTED** [review]

File: menu_trans.vim.template
Comment (flyingdragonfly) [inline menu_trans.vim.template]: Add a commented empty line above this line: `"`.

The translation in this file is not only for toolbar, but also menu items. It's easier that you simply say:

```
" Generated with the scripts from:
"     https://github.com/adaext/vim-menutrans-helper
```

File: menu_trans.vim.template
Comment (flyingdragonfly) [inline menu_trans.vim.template]: Let's keep the date updated until we finally merge this change.

**Review by flyingdragonfly — CHANGES_REQUESTED** [review]

File: vim_menutrans_merger.py
Comment (flyingdragonfly) [inline vim_menutrans_merger.py]: You're overwriting the template file, not the translation file.

**Review by flyingdragonfly — CHANGES_REQUESTED** [review]

**Review by flyingdragonfly — APPROVED** [review]

**Pull request merged**
